jsx/src/components/ServerDashboard/ServerDashboard.jsx

```diff
@@ -417,42 +443,39 @@ const ServerDashboard = (props) => {
               <th id="user-header">
                 User{" "}
                 <SortHandler
-                  sorts={{ asc: usernameAsc, desc: usernameDesc }}
-                  callback={(method) => setSortMethod(() => method)}
+                  currentSort={sort}
+                  setSort={setSort}
+                  sortKey="name"
                   testid="user-sort"
                 />
               </th>
-              <th id="admin-header">
-                Admin{" "}
-                <SortHandler
-                  sorts={{ asc: adminAsc, desc: adminDesc }}
-                  callback={(method) => setSortMethod(() => method)}
-                  testid="admin-sort"
-                />
-              </th>
-              <th id="server-header">
-                Server{" "}
-                <SortHandler
-                  sorts={{ asc: usernameAsc, desc: usernameDesc }}
-                  callback={(method) => setSortMethod(() => method)}
-                  testid="server-sort"
-                />
-              </th>
+              <th id="admin-header">Admin</th>
+              <th id="server-header">Server</th>
               <th id="last-activity-header">
                 Last Activity{" "}
                 <SortHandler
-                  sorts={{ asc: dateAsc, desc: dateDesc }}
-                  callback={(method) => setSortMethod(() => method)}
+                  currentSort={sort}
+                  setSort={setSort}
+                  sortKey="last_activity"
                   testid="last-activity-sort"
                 />
               </th>
               <th id="running-status-header">
-                Running{" "}
-                <SortHandler
-                  sorts={{ asc: runningAsc, desc: runningDesc }}
-                  callback={(method) => setSortMethod(() => method)}
-                  testid="running-status-sort"
-                />
+                <label title="only show active servers">
+                  <Form.Check
+                    inline
+                    type="checkbox"
+                    name="active_servers"
+                    aria-label="only show active servers"
+                    checked={state_filter == "active"}
+                    onChange={(event) => {
+                      setStateFilter(event.target.checked ? "active" : null);
+                    }}
+                  />
+                  {state_filter == "active"
+                    ? " Active servers"
+                    : " All servers"}
```

Comment:
![all-servers-active-servers](https://github.com/jupyterhub/jupyterhub/assets/3837114/12229ddd-73a1-4ed1-abc2-2e3f3f855410)

This didn't yet feel as intuitive as I think we can make it.

> - [ ] **All servers**

My thinking seeing this: "Ah so if I check this I can see all servers instead of a subset of them - what I'm currently seeing if not all servers then?"

> - [x] **Active servers**

My thinking seeing this: "Ah so if I uncheck this I can see not only the active servers!"

---

I think it would be better to just have `Active servers` at all time here.

> - [ ] **All servers**

For me,



This didn't feel intuitive for me, thinking the label should be Active servers at all times - where a unchecked "Active servers" checkbox means "you don't only see active servers, because this is unchecked"

Reply:
I think my suggestion is either:

```suggestion
                  " Active servers"
```

or

```suggestion
                  " Active servers filter"
```

---

With ` filter` added, I think it becomes a bit more clear that not having it checked will still include active servers - just that it also includes non-active servers as well. Not sure what I think is best among these choices.

Reply:
I struggled with that, too! A separate label makes sense, but I worried about using too much horizontal space in the header. 'Running' also doesn't make a huge amount of sense.

I put it in the same place the sort control used to be, but maybe that isn't the best idea.

WDYT about putting it next to the name filter? It's also a filter, so that makes sense to me. And there's plenty of room up there.

Reply:
Absolutely open to a relocation of checkbox+label, I'm not certain if I'd like it better or not - the key UX confusion point I perceived was the changing checkbox label. I think having it close to name filtering probably is good, so hmmm i'm :+1: for a relocation also!